packages/bruno-common/src/interpolate/index.ts

```diff
@@ -11,21 +11,46 @@
  * Output: Hello, my name is Bruno and I am 4 years old
  */
 
+import { Set } from 'typescript';
 import { flattenObject } from '../utils';
 
 const interpolate = (str: string, obj: Record<string, any>): string => {
   if (!str || typeof str !== 'string' || !obj || typeof obj !== 'object') {
     return str;
   }
 
-  const patternRegex = /\{\{([^}]+)\}\}/g;
   const flattenedObj = flattenObject(obj);
-  const result = str.replace(patternRegex, (match, placeholder) => {
+
+  return replace(str, flattenedObj);
+};
+
+const replace = (
+  str: string,
+  flattenedObj: Record<string, any>,
+  visited = new Set<String>(),
+  results = new Map<string, string>()
+): string => {
+  const patternRegex = /\{\{([^}]+)\}\}/g;
+
+  return str.replace(patternRegex, (match, placeholder) => {
     const replacement = flattenedObj[placeholder];
-    return replacement !== undefined ? replacement : match;
-  });
 
-  return result;
+    if (results.has(match)) {
+      return results.get(match);
+    } else if (patternRegex.test(replacement) && !visited.has(match)) {
```

Comment:
No need for an else since you return in the previous if

Reply:
Dene